### PR TITLE
Added React week4 homework

### PR DESCRIPTION
**Pull request by sandhana14**

### CODE REVIEW

File: react/week4/github-user/src/GithubRepos.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+const HackYourFutureRepos = () => {
+  const [repos, setrepos] = useState([]);
+
+  const hyfRepos = () => {
+    fetch("https://api.github.com/users/HackYourFuture-CPH/repos")
+      .then((response) => response.json())
+      .then((repos) => {
+        console.log(repos);
+        return setrepos(repos);
+      });
```
Comment (obelitz) [inline react/week4/github-user/src/GithubRepos.js:12]: If you want to shorten this, you can return with the fat arrow here as well:
`.then((repos) => setRepos(repos))`

File: react/week4/github-user/src/GithubRepos.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  };
+
+  return (
+    <div>
+      <h1 onClick={hyfRepos}>Hack your future- CPh Repos</h1>
+      {repos.map((repo) => (
+        <div key={repo.id}>{repo.name}</div>
+      ))}
```
Comment (obelitz) [inline react/week4/github-user/src/GithubRepos.js:20]: Remember to make it obvious to the user how they can get the repos. 
Either with a button "Get repos" or by changing the title to "Click here to see HYF Repos"

File: react/week4/github-user/src/GithubRepos.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { useState } from "react";
+
+const HackYourFutureRepos = () => {
+  const [repos, setrepos] = useState([]);
+
+  const hyfRepos = () => {
```
Comment (obelitz) [inline react/week4/github-user/src/GithubRepos.js:6]: Method names for methods like this one usually begin with "fetch" or "get".

File: react/week4/github-user/src/FetchDataContext.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
+import { useContext, useEffect } from "react";
+import { GithubUserContext } from "./GithubStates";
+
+const FetchDataContext = () => {
```
Comment (obelitz) [inline react/week4/github-user/src/FetchDataContext.js:4]: This doesn't seem to be a Context. This is more of a `UsersList`

File: react/week4/github-user/src/FetchDataContext.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  const GithubUserContextValue = useContext(GithubUserContext);
+
+  const onChangeInputHandle = (event) => {
+    event.preventDefault();
+    GithubUserContextValue.setInputValue(event.target.value);
+  };
+
+  const githubFetchApi = () => {
```
Comment (obelitz) [inline react/week4/github-user/src/FetchDataContext.js:12]: I'd call this method something more action-y like `searchGithubUsers` or `fetchUsers`

File: react/week4/github-user/src/FetchDataContext.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    if (GithubUserContextValue.inputValue) {
+      GithubUserContextValue.setIsLoading(true);
+      fetch(
+        `https://api.github.com/search/users?q=${GithubUserContextValue.inputValue}`
+      )
+        .then((response) => response.json())
+        .then((data) => {
+          GithubUserContextValue.setIsLoading(false);
+          if (data.message) {
+            GithubUserContextValue.setmessage(data.message);
```
Comment (obelitz) [inline react/week4/github-user/src/FetchDataContext.js:22]: Remember to reset the message on every new search, so the user doesn't see both the message and the results :)

File: react/week4/github-user/src/FetchDataContext.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+          } else if (data.items.length) {
+            GithubUserContextValue.setUsers(data.items);
+          } else {
+            GithubUserContextValue.setmessage("No results......");
+          }
+        });
```
Comment (obelitz) [inline react/week4/github-user/src/FetchDataContext.js:28]: When working with promises, it's generally nice to put a `.catch` at the end to handle unexpected errors

**Review by obelitz — COMMENTED** [review]
Good job!
Just had a few comments on some naming and some UI :)

File: react/week4/github-user/src/FetchDataContext.js (hunk 7 above)
Comment (sandhana14) [reply react/week4/github-user/src/FetchDataContext.js:28]: Thank you so much for your comments. I will make the changes as per your comments.

**Pull request merged**